File: app/src/main/java/me/kirimin/mitsumine/ui/activity/TopActivity.java
package me.kirimin.mitsumine.ui.activity;

import butterknife.ButterKnife;
import butterknife.InjectView;
import me.kirimin.mitsumine.db.AccountDAO;
import me.kirimin.mitsumine.db.FeedDAO;
import me.kirimin.mitsumine.db.KeywordDAO;
import me.kirimin.mitsumine.db.UserIdDAO;
import me.kirimin.mitsumine.model.Account;
import me.kirimin.mitsumine.network.api.FeedApi.CATEGORY;
import me.kirimin.mitsumine.network.api.FeedApi.TYPE;
import me.kirimin.mitsumine.ui.activity.search.KeywordSearchActivity;
import me.kirimin.mitsumine.ui.activity.search.MyBookmarksActivity;
import me.kirimin.mitsumine.ui.activity.search.UserSearchActivity;
import me.kirimin.mitsumine.ui.fragment.FeedFragment;

import me.kirimin.mitsumine.R;

import android.app.AlertDialog;
import android.content.DialogInterface;
import android.content.Intent;
import android.content.res.Configuration;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v7.app.ActionBarDrawerToggle;
import android.support.v4.view.MenuItemCompat;
import android.support.v4.widget.DrawerLayout;
import android.support.v7.app.ActionBar;
import android.support.v7.app.ActionBarActivity;
import android.support.v7.widget.Toolbar;
import android.view.Gravity;
import android.view.LayoutInflater;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.view.View.OnClickListener;
import android.view.View.OnLongClickListener;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.makeramen.RoundedTransformationBuilder;
import com.squareup.picasso.Picasso;
import com.squareup.picasso.Transformation;

public class TopActivity extends ActionBarActivity implements ActionBar.OnNavigationListener {

    @InjectView(R.id.drawerLayout)
    DrawerLayout mDrawerLayout;
    @InjectView(R.id.MainNavigationReadTextView)
    TextView navigationReadTextView;
    @InjectView(R.id.MainNavigationSettingsTextView)
    TextView navigationSettingTextView;
    @InjectView(R.id.MainNavigationKeywordSearchTextView)
    TextView navigationKeywordSearchTextView;
    @InjectView(R.id.MainNavigationUserSearchTextView)
    TextView navigationUserSearchTextView;
    @InjectView(R.id.MainNavigationCategories)
    LinearLayout navigationCategoriesLayout;
    @InjectView(R.id.MainNavigationAdditions)
    LinearLayout navigationAdditionsLayout;
    @InjectView(R.id.MainNavigationLoginButton)
    Button navigationLoginButton;
    @InjectView(R.id.MainNavigationUserInfoLayout)
    View navigationUserInfoLayout;
    @InjectView(R.id.MainNavigationUserName)
    TextView navigationUserNameTextView;
    @InjectView(R.id.MainNavigationUserIconImageView)
    ImageView navigationUserIconImageView;

    private ActionBarDrawerToggle mDrawerToggle;
    private CATEGORY mSelectedCategory;
    private TYPE mSelectedType;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_top);
        ButterKnife.inject(this);

        // 古い既読を削除
        FeedDAO.deleteOldData();

        Toolbar toolbar = (Toolbar) findViewById(R.id.tool_bar);
        setSupportActionBar(toolbar);
        ActionBar actionBar = getSupportActionBar();

        actionBar.setNavigationMode(ActionBar.NAVIGATION_MODE_LIST);
        actionBar.setDisplayHomeAsUpEnabled(true);
        actionBar.setHomeButtonEnabled(true);

        String[] data = new String[]{getString(R.string.feed_hot), getString(R.string.feed_new)};
        ArrayAdapter<String> adapter = new ArrayAdapter<>(actionBar.getThemedContext(), android.R.layout.simple_list_item_1, data);
        actionBar.setListNavigationCallbacks(adapter, this);

        mDrawerToggle = new ActionBarDrawerToggle(this, mDrawerLayout, R.string.app_name, R.string.app_name);
        mDrawerLayout.setDrawerListener(mDrawerToggle);
        toolbar.setOnClickListener(new OnClickListener() {
            @Override
            public void onClick(View v) {
                if (mDrawerLayout.isDrawerOpen(Gravity.LEFT)) {
                    mDrawerLayout.closeDrawer(Gravity.LEFT);
                } else {
                    mDrawerLayout.openDrawer(Gravity.LEFT);
                }
            }
        });
        navigationReadTextView.setOnClickListener(new OnClickListener() {

            @Override
            public void onClick(View v) {
                startActivity(new Intent(TopActivity.this, ReadActivity.class));
                mDrawerLayout.closeDrawers();
            }
        });
        navigationSettingTextView.setOnClickListener(new OnClickListener() {

            @Override
            public void onClick(View v) {
                startActivity(new Intent(TopActivity.this, SettingsActivity.class));
                mDrawerLayout.closeDrawers();
            }
        });
        navigationKeywordSearchTextView.setOnClickListener(new OnClickListener() {

            @Override
            public void onClick(View v) {
                startActivity(new Intent(TopActivity.this, KeywordSearchActivity.class));
                mDrawerLayout.closeDrawers();
            }
        });
        navigationUserSearchTextView.setOnClickListener(new OnClickListener() {

            @Override
            public void onClick(View v) {
                startActivity(new Intent(TopActivity.this, UserSearchActivity.class));
                mDrawerLayout.closeDrawers();
            }
        });
        navigationUserInfoLayout.setOnClickListener(new OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(TopActivity.this, MyBookmarksActivity.class);
                intent.putExtras(MyBookmarksActivity.buildBundle(""));
                startActivity(intent);
            }
        });
        navigationLoginButton.setOnClickListener(new OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(TopActivity.this, LoginActivity.class));
            }
        });

        navigationCategoriesLayout.addView(makeNavigationCategoryButton(getString(R.string.feed_main), CATEGORY.MAIN));
        navigationCategoriesLayout.addView(makeNavigationCategoryButton(getString(R.string.feed_social), CATEGORY.SOCIAL));
        navigationCategoriesLayout.addView(makeNavigationCategoryButton(getString(R.string.feed_economics), CATEGORY.ECONOMICS));
        navigationCategoriesLayout.addView(makeNavigationCategoryButton(getString(R.string.feed_life), CATEGORY.LIFE));
        navigationCategoriesLayout.addView(makeNavigationCategoryButton(getString(R.string.feed_knowledge), CATEGORY.KNOWLEDGE));
        navigationCategoriesLayout.addView(makeNavigationCategoryButton(getString(R.string.feed_it), CATEGORY.IT));
        navigationCategoriesLayout.addView(makeNavigationCategoryButton(getString(R.string.feed_fun), CATEGORY.FUN));
        navigationCategoriesLayout.addView(makeNavigationCategoryButton(getString(R.string.feed_entertainment), CATEGORY.ENTERTAINMENT));
        navigationCategoriesLayout.addView(makeNavigationCategoryButton(getString(R.string.feed_game), CATEGORY.GAME));
        changeShowCategory(getString(R.string.feed_main), CATEGORY.MAIN);
    }

    @Override
    protected void onStart() {
        super.onStart();
        loadNavigationButtons();
        Account account = AccountDAO.get();
        if (account != null) {
            navigationUserInfoLayout.setVisibility(View.VISIBLE);
            navigationLoginButton.setVisibility(View.GONE);
            navigationUserNameTextView.setText(account.urlName);
            Transformation transformation = new RoundedTransformationBuilder().borderWidthDp(0).cornerRadiusDp(48).oval(false).build();
            Picasso.with(this).load(account.imageUrl).transform(transformation).fit().into(navigationUserIconImageView);
        } else {
            navigationUserInfoLayout.setVisibility(View.GONE);
            navigationLoginButton.setVisibility(View.VISIBLE);
        }
    }

    @Override
    public boolean onNavigationItemSelected(int itemPosition, long itemId) {
        mDrawerLayout.closeDrawers();
        Fragment fragment;
        if (itemPosition == 0) {
            fragment = FeedFragment.newFragment(mSelectedCategory, TYPE.HOT);
            mSelectedType = TYPE.HOT;
        } else if (itemPosition == 1) {
            fragment = FeedFragment.newFragment(mSelectedCategory, TYPE.NEW);
            mSelectedType = TYPE.NEW;
        } else {
            return true;
        }
        getSupportFragmentManager().beginTransaction()
                .replace(R.id.containerFrameLayout, fragment)
                .commit();
        return true;
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        MenuItem unRead = menu.add(Menu.NONE, 1, Menu.NONE, R.string.type_read_later);
        unRead.setIcon(R.drawable.ic_action_labels);
        MenuItemCompat.setShowAsAction(unRead, MenuItemCompat.SHOW_AS_ACTION_ALWAYS);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if (mDrawerToggle.onOptionsItemSelected(item)) {
            return true;
        }

        if (item.getTitle().equals(getString(R.string.type_read_later))) {
            startActivity(new Intent(this, ReadLaterActivity.class));
        }
        return super.onOptionsItemSelected(item);
    }

    @Override
    protected void onPostCreate(Bundle savedInstanceState) {
        super.onPostCreate(savedInstanceState);
        mDrawerToggle.syncState();
    }

    @Override
    public void onConfigurationChanged(Configuration newConfig) {
        super.onConfigurationChanged(newConfig);
        mDrawerToggle.onConfigurationChanged(newConfig);
    }

    @Override
    public void onBackPressed() {
        if (mDrawerLayout.isDrawerOpen(Gravity.LEFT)) {
            mDrawerLayout.closeDrawer(Gravity.LEFT);
        } else {
            super.onBackPressed();
        }
    }

    private void loadNavigationButtons() {
        navigationAdditionsLayout.removeAllViews();
        for (final String keyword : KeywordDAO.findAll()) {
            navigationAdditionsLayout.addView(makeNavigationButton(keyword, new OnClickListener() {

                @Override
                public void onClick(View v) {
                    Intent intent = new Intent(TopActivity.this, KeywordSearchActivity.class);
                    startActivity(intent.putExtras(KeywordSearchActivity.buildBundle(keyword)));
                    mDrawerLayout.closeDrawers();
                }
            }, new OnLongClickListener() {

                @Override
                public boolean onLongClick(View v) {
                    buildDeleteDialog(keyword, v).show();
                    return false;
                }
            }));
        }
        for (final String userId : UserIdDAO.findAll()) {
            navigationAdditionsLayout.addView(makeNavigationButton(userId, new OnClickListener() {

                @Override
                public void onClick(View v) {
                    Intent intent = new Intent(TopActivity.this, UserSearchActivity.class);
                    startActivity(intent.putExtras(UserSearchActivity.buildBundle(userId)));
                    mDrawerLayout.closeDrawers();
                }
            }, new OnLongClickListener() {

                @Override
                public boolean onLongClick(View v) {
                    buildDeleteUserIdDialog(userId, v).show();
                    return false;
                }
            }));
        }
    }

    private View makeNavigationCategoryButton(final String label, final CATEGORY category) {
        return makeNavigationButton(label, new OnClickListener() {

            @Override
            public void onClick(View v) {
                mDrawerLayout.closeDrawers();
                changeShowCategory(label, category);
            }
        }, null);
    }

    private View makeNavigationButton(String label, OnClickListener onClick, OnLongClickListener onLongClick) {
        View navigationView = LayoutInflater.from(getApplicationContext()).inflate(R.layout.activity_top_navigation, null);
        TextView textView = (TextView) navigationView.findViewById(R.id.MainNavigationTextView);
        textView.setText(label);
        textView.setOnClickListener(onClick);
        textView.setOnLongClickListener(onLongClick);
        return navigationView;
    }

    private void changeShowCategory(final String label, final CATEGORY category) {
        mSelectedCategory = category;
        getSupportActionBar().setTitle(label);
        getSupportFragmentManager().beginTransaction()
                .replace(R.id.containerFrameLayout, FeedFragment.newFragment(category, mSelectedType))
                .commit();
        for (int i = 0; i < navigationCategoriesLayout.getChildCount(); i++) {
            TextView categoryButton = (TextView) navigationCategoriesLayout.getChildAt(i).findViewById(R.id.MainNavigationTextView);
            categoryButton.setTextColor(getResources().getColor(categoryButton.getText().equals(label) ? R.color.orange : R.color.text));
        }
    }

    private AlertDialog buildDeleteDialog(final String word, final View view) {
        DialogInterface.OnClickListener onPositiveListener = new DialogInterface.OnClickListener() {

            @Override
            public void onClick(DialogInterface dialog, int which) {
                KeywordDAO.delete(word);
                view.setVisibility(View.GONE);
            }
        };

        return new AlertDialog.Builder(this)
                .setTitle(R.string.settings_ngword_delete)
                .setPositiveButton(android.R.string.ok, onPositiveListener)
                .setNegativeButton(android.R.string.cancel, null)
                .create();
    }

    private AlertDialog buildDeleteUserIdDialog(final String word, final View view) {
        DialogInterface.OnClickListener onPositiveListener = new DialogInterface.OnClickListener() {

            @Override
            public void onClick(DialogInterface dialog, int which) {
                UserIdDAO.delete(word);
                view.setVisibility(View.GONE);
            }
        };

        return new AlertDialog.Builder(this)
                .setTitle(R.string.settings_ngword_delete)
                .setPositiveButton(android.R.string.ok, onPositiveListener)
                .setNegativeButton(android.R.string.cancel, null)
                .create();
    }
}

File: app/src/main/java/me/kirimin/mitsumine/ui/activity/TopActivity.kt
package me.kirimin.mitsumine.ui.activity

import me.kirimin.mitsumine.db.AccountDAO
import me.kirimin.mitsumine.db.FeedDAO
import me.kirimin.mitsumine.db.KeywordDAO
import me.kirimin.mitsumine.db.UserIdDAO
import me.kirimin.mitsumine.model.Account
import me.kirimin.mitsumine.network.api.FeedApi.CATEGORY
import me.kirimin.mitsumine.network.api.FeedApi.TYPE
import me.kirimin.mitsumine.ui.activity.search.KeywordSearchActivity
import me.kirimin.mitsumine.ui.activity.search.MyBookmarksActivity
import me.kirimin.mitsumine.ui.activity.search.UserSearchActivity
import me.kirimin.mitsumine.ui.fragment.FeedFragment

import me.kirimin.mitsumine.R

import android.app.AlertDialog
import android.content.DialogInterface
import android.content.Intent
import android.content.res.Configuration
import android.os.Bundle
import android.support.v7.app.ActionBarDrawerToggle
import android.support.v4.view.MenuItemCompat
import android.support.v4.widget.DrawerLayout
import android.support.v7.app.ActionBar
import android.support.v7.app.ActionBarActivity
import android.support.v7.widget.Toolbar
import android.view.Gravity
import android.view.LayoutInflater
import android.view.Menu
import android.view.MenuItem
import android.view.View
import android.view.View.OnClickListener
import android.view.View.OnLongClickListener
import android.widget.ArrayAdapter
import android.widget.Button
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.TextView

import com.makeramen.RoundedTransformationBuilder
import com.squareup.picasso.Picasso
import com.squareup.picasso.Transformation

import kotlinx.android.synthetic.activity_top.*
import me.kirimin.mitsumine.ui.activity.search.SearchActivity
import java.io.Serializable

public class TopActivity : ActionBarActivity() {

    private var mDrawerToggle: ActionBarDrawerToggle? = null
    private var mSelectedCategory = CATEGORY.MAIN
    private var mSelectedType = TYPE.HOT

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_top)

        // 古い既読を削除
        FeedDAO.deleteOldData()

        setSupportActionBar(toolBar)
        val actionBar = getSupportActionBar()

        actionBar.setNavigationMode(ActionBar.NAVIGATION_MODE_LIST)
        actionBar.setDisplayHomeAsUpEnabled(true)
        actionBar.setHomeButtonEnabled(true)

        val data = arrayOf(getString(R.string.feed_hot), getString(R.string.feed_new))
        val adapter = ArrayAdapter(actionBar.getThemedContext(), android.R.layout.simple_list_item_1, data)
        actionBar.setListNavigationCallbacks(adapter, { position, id ->
            drawerLayout.closeDrawers()
            mSelectedType = if (position == 0) TYPE.HOT else TYPE.NEW
            refreshShowCategoryAndType()
            true
        })

        mDrawerToggle = ActionBarDrawerToggle(this, drawerLayout, R.string.app_name, R.string.app_name)
        drawerLayout.setDrawerListener(mDrawerToggle)
        toolBar.setOnClickListener {
            if (drawerLayout.isDrawerOpen(Gravity.LEFT)) {
                drawerLayout.closeDrawer(Gravity.LEFT)
            } else {
                drawerLayout.openDrawer(Gravity.LEFT)
            }
        }

        navigationReadTextView.setOnClickListener {
            startActivity(Intent(this, javaClass<ReadActivity>()))
            drawerLayout.closeDrawers()
        }
        navigationSettingsTextView.setOnClickListener {
            startActivity(Intent(this, javaClass<SettingActivity>()))
            drawerLayout.closeDrawers()
        }
        navigationKeywordSearchTextView.setOnClickListener {
            startActivity(Intent(this, javaClass<KeywordSearchActivity>()))
            drawerLayout.closeDrawers()
        }
        navigationUserSearchTextView.setOnClickListener {
            startActivity(Intent(this, javaClass<UserSearchActivity>()))
            drawerLayout.closeDrawers()
        }
        navigationUserInfoLayout.setOnClickListener {
            val intent = Intent(this, javaClass<MyBookmarksActivity>())
            intent.putExtras(SearchActivity.buildBundle("", false))
            startActivity(intent)
        }
        navigationLoginButton.setOnClickListener {
            startActivity(Intent(this, javaClass<LoginActivity>()))
        }

        navigationCategories.addView(makeNavigationCategoryButton(CATEGORY.MAIN))
        navigationCategories.addView(makeNavigationCategoryButton(CATEGORY.SOCIAL))
        navigationCategories.addView(makeNavigationCategoryButton(CATEGORY.ECONOMICS))
        navigationCategories.addView(makeNavigationCategoryButton(CATEGORY.LIFE))
        navigationCategories.addView(makeNavigationCategoryButton(CATEGORY.KNOWLEDGE))
        navigationCategories.addView(makeNavigationCategoryButton(CATEGORY.IT))
        navigationCategories.addView(makeNavigationCategoryButton(CATEGORY.FUN))
        navigationCategories.addView(makeNavigationCategoryButton(CATEGORY.ENTERTAINMENT))
        navigationCategories.addView(makeNavigationCategoryButton(CATEGORY.GAME))

        savedInstanceState?.let {
            mSelectedCategory = savedInstanceState?.getSerializable(javaClass<CATEGORY>().getCanonicalName()) as CATEGORY
            mSelectedType = savedInstanceState?.getSerializable(javaClass<TYPE>().getCanonicalName()) as TYPE
        }
        refreshShowCategoryAndType()
    }

    override fun onStart() {
        super.onStart()
        loadNavigationButtons()
        val account = AccountDAO.get()
        if (account != null) {
            navigationUserInfoLayout.setVisibility(View.VISIBLE)
            navigationLoginButton.setVisibility(View.GONE)
            navigationUserName.setText(account.urlName)
            val transformation = RoundedTransformationBuilder().borderWidthDp(0f).cornerRadiusDp(48f).oval(false).build()
            Picasso.with(this).load(account.imageUrl).transform(transformation).fit().into(navigationUserIconImageView)
        } else {
            navigationUserInfoLayout.setVisibility(View.GONE)
            navigationLoginButton.setVisibility(View.VISIBLE)
        }
    }

    override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)
        outState.putSerializable(javaClass<CATEGORY>().getCanonicalName(), mSelectedCategory as Serializable)
        outState.putSerializable(javaClass<TYPE>().getCanonicalName(), mSelectedType as Serializable)
    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        val unRead = menu!!.add(Menu.NONE, 1, Menu.NONE, R.string.type_read_later)
        unRead.setIcon(R.drawable.ic_action_labels)
        MenuItemCompat.setShowAsAction(unRead, MenuItemCompat.SHOW_AS_ACTION_ALWAYS)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem?): Boolean {
        if (mDrawerToggle!!.onOptionsItemSelected(item)) {
            return true
        }

        if (item!!.getTitle() == getString(R.string.type_read_later)) {
            startActivity(Intent(this, javaClass<ReadLaterActivity>()))
        }
        return super.onOptionsItemSelected(item)
    }

    override fun onPostCreate(savedInstanceState: Bundle?) {
        super.onPostCreate(savedInstanceState)
        mDrawerToggle!!.syncState()
    }

    override fun onConfigurationChanged(newConfig: Configuration) {
        super.onConfigurationChanged(newConfig)
        mDrawerToggle!!.onConfigurationChanged(newConfig)
    }

    override fun onBackPressed() {
        if (drawerLayout.isDrawerOpen(Gravity.LEFT)) {
            drawerLayout.closeDrawer(Gravity.LEFT)
        } else {
            super.onBackPressed()
        }
    }

    private fun loadNavigationButtons() {
        navigationAdditions.removeAllViews()
        KeywordDAO.findAll().forEach { keyword ->
            navigationAdditions.addView(makeNavigationButton(keyword,
                    OnClickListener {
                        val intent = Intent(this@TopActivity, javaClass<KeywordSearchActivity>())
                        startActivity(intent.putExtras(SearchActivity.buildBundle(keyword)))
                        drawerLayout.closeDrawers()
                    },
                    OnLongClickListener { v ->
                        buildDeleteDialog(keyword, v).show()
                        false
                    }))
        }
        UserIdDAO.findAll().forEach { userId ->
            navigationAdditions.addView(makeNavigationButton(userId,
                    OnClickListener {
                        val intent = Intent(this@TopActivity, javaClass<UserSearchActivity>())
                        startActivity(intent.putExtras(SearchActivity.buildBundle(userId)))
                        drawerLayout.closeDrawers()
                    },
                    OnLongClickListener { v ->
                        buildDeleteUserIdDialog(userId, v).show()
                        false
                    }))
        }
    }

    private fun makeNavigationCategoryButton(category: CATEGORY): View {
        return makeNavigationButton(getString(category.labelResource), OnClickListener { v ->
            drawerLayout.closeDrawers()
            mSelectedCategory = category
            refreshShowCategoryAndType()
        }, null)
    }

    private fun makeNavigationButton(label: String, onClick: OnClickListener, onLongClick: OnLongClickListener?): View {
        val navigationView = LayoutInflater.from(getApplicationContext()).inflate(R.layout.activity_top_navigation, null)
        val textView = navigationView.findViewById(R.id.MainNavigationTextView) as TextView
        textView.setText(label)
        textView.setOnClickListener(onClick)
        textView.setOnLongClickListener(onLongClick)
        return navigationView
    }

    private fun refreshShowCategoryAndType() {
        getSupportActionBar().setTitle(mSelectedCategory.labelResource)
        getSupportActionBar().setSelectedNavigationItem(if (mSelectedType == TYPE.HOT) 0 else 1)
        getSupportFragmentManager().beginTransaction().replace(R.id.containerFrameLayout, FeedFragment.newFragment(mSelectedCategory, mSelectedType)).commit()
        for (i in 0..navigationCategories.getChildCount() - 1) {
            val categoryButton = navigationCategories.getChildAt(i).findViewById(R.id.MainNavigationTextView) as TextView
            val isSelectedCategory = categoryButton.getText() == getString(mSelectedCategory.labelResource)
            categoryButton.setTextColor(getResources().getColor(if (isSelectedCategory) R.color.orange else R.color.text))
        }
    }

    private fun buildDeleteDialog(word: String, view: View): AlertDialog {
        return AlertDialog.Builder(this)
                .setTitle(R.string.settings_ngword_delete)
                .setPositiveButton(android.R.string.ok, { dialog, id ->
                    KeywordDAO.delete(word)
                    view.setVisibility(View.GONE)
                })
                .setNegativeButton(android.R.string.cancel, null)
                .create()
    }

    private fun buildDeleteUserIdDialog(word: String, view: View): AlertDialog {
        return AlertDialog.Builder(this)
                .setTitle(R.string.settings_ngword_delete)
                .setPositiveButton(android.R.string.ok, { dialog, id ->
                    UserIdDAO.delete(word)
                    view.setVisibility(View.GONE)
                })
                .setNegativeButton(android.R.string.cancel, null).create()
    }
}
